cclib/method/bader.py

```diff
@@ -0,0 +1,223 @@
+
+                            nextDirection = list(zip(maxat[0], maxat[1], maxat[2]))[0]
+                            nextDirection = [ind - 1 for ind in nextDirection]
+
+                            if len(list(zip(maxat[0], maxat[1], maxat[2]))) > 1:
+                                # when one or more directions indicate max grad (of 0), prioritize
+                                # to include all points in the Bader space
+                                if list(zip(maxat[0], maxat[1], maxat[2]))[0] == (1, 1, 1,):
+                                    nextDirection = list(zip(maxat[0], maxat[1], maxat[2]))[1]
+                                    nextDirection = [ind - 1 for ind in nextDirection]
+
+                            listcoord.append((xindex, yindex, zindex))
+
+                            if (
+                                self.fragresults[
+                                    xindex + nextDirection[0],
+                                    yindex + nextDirection[1],
+                                    zindex + nextDirection[2],
+                                ]
+                                != 0
+                            ):
+                                # Path arrived at a point that has already been assigned with an index
+                                baderIndex = self.fragresults[
+                                    xindex + nextDirection[0],
+                                    yindex + nextDirection[1],
+                                    zindex + nextDirection[2],
+                                ]
+                                listcoord = tuple(numpy.array(listcoord).T)
+                                self.fragresults[listcoord] = baderIndex
+
+                                loopcondition = False
+
+                            elif nextDirection == [0, 0, 0]:
+                                # This point is local maximum
+                                # Assign this point and everything along the path with a new index
+                                baderIndex = nextIndex
+                                nextIndex += 1
+
+                                listcoord = tuple(numpy.array(listcoord).T)
+                                self.fragresults[listcoord] = baderIndex
+
+                                loopcondition = False
+
+                            elif (
+                                xindex + nextDirection[0] == 0
+                                or xindex + nextDirection[0] == (len(self.chgdensity.data) - 1)
+                                or yindex + nextDirection[1] == 0
+                                or yindex + nextDirection[1] == (len(self.chgdensity.data[0]) - 1)
+                                or zindex + nextDirection[2] == 0
+                                or zindex + nextDirection[2]
+                                == (len(self.chgdensity.data[0][0]) - 1)
+                            ):
+                                # Avoid exiting the grid
```

Comment:
The body of this conditional is the same as the previous conditional, let's combine them.

Reply:
I've combined them. Thanks!